client/src/components/squids/SquidList.jsx

```diff
@@ -12,32 +13,42 @@ export const SquidList = () => {
   const [pageNumber, setPageNumber] = useState(1);
   const [pageCount, setPageCount] = useState(1);
 
-  const { data = {}, isLoading } = useSquidList({ pageIndex: pageNumber - 1, itemsPerPage });
-
-  const squidData = data.squids || [];
+  const {
+    data,
+    isLoading,
+    refetch: refetchSquids,
+  } = useSquidList({ pageIndex: pageNumber - 1, itemsPerPage });
 
   useEffect(() => {
-    setPageCount(data.pageCount);
+    setPageCount(data?.pageCount);
   }, [data]);
 
-  const squids = squidData.map((squid) => <SquidInfo key={squid.id} {...squid} />);
-
   const handlePageClick = (event) => {
     setPageNumber(event.selected + 1);
   };
 
-  let pageHeader;
+  let pageHeader = <p className="squidList__heading">(page 1 / 1)</p>;
 
   if (!isLoading) {
     pageHeader = <p className="squidList__heading">{`(page ${pageNumber} / ${pageCount})`}</p>;
   }
 
+  const squidsData = data?.squids || [];
+  const specialPowers = data?.specialPowers || [];
```

Comment:
Should we separate this GET request into two: one for the squid data and one for the squid metadata (here, the special powers)? It would let you easily separate these sets of data in the cache and potentially reuse the hooks later in other parts of the app

Reply:
Definitely, that makes sense for reusability and separating concerns! I was thinking of reducing the number of requests on the page, but the cached data helps